t/43-scheduling-and-worker-scalability.t

```diff
@@ -0,0 +1,194 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/lib";
+use OpenQA::Scheduler::Model::Jobs;
+use OpenQA::Test::Database;
+use OpenQA::Jobs::Constants;
+use OpenQA::Test::Utils
+  qw(create_user_for_workers create_webapi setup_share_dir create_websocket_server),
+  qw(stop_service setup_fullstack_temp_dir);
+use Test::More;
+use Test::Warnings;
+use Time::HiRes 'sleep';
+use File::Path 'make_path';
+use Scalar::Util 'looks_like_number';
+use Mojo::File 'path';
+use Mojo::Util 'dumper';
+
+# allow the scheduler to assigns all jobs within one tick (needs to be in BEGIN block because the env variable
+# is assigned to constant)
+BEGIN {
+    $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} = $ENV{SCALABILITY_FULLSTACK_JOB_COUNT}
+      // $ENV{SCALABILITY_FULLSTACK_WORKER_COUNT};
+}
+
+# read number of workers to spawn from environment variable; skip test entirely if variable not present
+# similar to other fullstack tests
+my $worker_count = $ENV{SCALABILITY_FULLSTACK_WORKER_COUNT};
+my $job_count    = $ENV{SCALABILITY_FULLSTACK_JOB_COUNT} // $worker_count;
+plan skip_all => 'set e.g. SCALABILITY_FULLSTACK_WORKER_COUNT=n to run the scalability test for n workers'
+  unless looks_like_number($worker_count) && $worker_count > 0;
+
+# setup basedir, config dir and database
+my $tempdir = setup_fullstack_temp_dir('scalability');
+my $schema  = OpenQA::Test::Database->new->create(skip_fixtures => 1);
+my $workers = $schema->resultset('Workers');
+my $jobs    = $schema->resultset('Jobs');
+
+# create web UI and websocket server
+my $mojoport              = Mojo::IOLoop::Server->generate_port();
+my $web_socket_server_pid = create_websocket_server($mojoport + 1, 0, 1, 1);
+my $webui_pid             = create_webapi($mojoport, sub { });
+
+# prepare spawning workers
+my $sharedir        = setup_share_dir($ENV{OPENQA_BASEDIR});
+my $resultdir       = path($ENV{OPENQA_BASEDIR}, 'openqa', 'testresults')->make_path;
+my $api_credentials = create_user_for_workers;
+my $api_key         = $api_credentials->key;
+my $api_secret      = $api_credentials->secret;
+my $webui_host      = "http://localhost:$mojoport";
+my $worker_path     = path($FindBin::Bin)->child('../script/worker');
+my $isotovideo_path = path($FindBin::Bin)->child('dummy-isotovideo.sh');
+my $worker_flags    = '--verbose --no-cleanup';
+my $worker_args
+  = "--apikey=$api_key --apisecret=$api_secret --host=$webui_host --isotovideo=$isotovideo_path $worker_flags";
+note("share dir: $sharedir");
+note("resultdir: $resultdir");
+
+# spawn workers
+note("Spawning $worker_count workers");
+sub spawn_worker {
+    my ($instance) = @_;
+
+    note("Starting worker '$instance'");
+    my $workerpid = fork();
+    return $workerpid if $workerpid != 0;
+
+    exec("perl $worker_path --instance=$instance $worker_args");
```

Comment:
As I learned yesterday I think it's better to use a list of command and arguments as described on https://perldoc.perl.org/functions/exec.html, i.e.

```perl
    exec('perl', $worker_path, "--instance=$instance", $worker_args);
```